paddle/fluid/framework/garbage_collector.cc

```diff
@@ -28,8 +28,14 @@
 namespace paddle {
 namespace framework {
 
+#ifdef PADDLE_ON_INFERENCE
+static const double kDefaultEagerDeleteTensorGB = -1;
```

Comment:
Since the code of gc is embedded in `Executor`, which may affect sub_block ops in inference. Disable it when `PADDLE_ON_INFERENCE` is defined.